test/test_ops.py

```diff
@@ -37,12 +37,10 @@ def test_unsupported_dtypes(self, device, dtype, op):
         # sample_inputs can have a function for generating the input that doesn't work for specified dtype
         # https://github.com/pytorch/pytorch/issues/49024
         with self.assertRaises(RuntimeError):
-            samples = op.sample_inputs(device, dtype)
-            if len(samples) == 0:
-                self.skipTest("Skipped! No sample inputs!")
-
             # NOTE: only tests on first sample
-            sample = samples[0]
+            sample = op.get_one_sample_input(device, dtype)
```

Comment:
For code velocity let's limit the application of this technique to this test for now (if we end up with a pattern where tests have to change at all), that way all the tests won't need to change and we can focus on the design

Reply:
to get a good feedback of how the API will look like, we need at least a test which uses `get_one_sample_input` pattern and one which uses the `for sample in samples:` pattern. But indeed, it sounds like a good idea, I'll do it